klaxer/rules.py

```diff
@@ -0,0 +1,207 @@
+from klaxer.models import Severity
+from klaxer.errors import ServiceNotDefinedError, ConfigurationError
+
+
+class Rules:
+    def __init__(self):
+        self._classification_rules = {}
+        self._exclusion_rules = {}
+        self._enrichment_rules = {}
+        self._routing_rules = {}
+        self._config = None
+
+        try:
+            # TODO: Absolute path? Where should this live?
+            with open('config/klaxer.yml', 'r') as ymlfile:
+                self._config = yaml.load(ymlfile)
+        except yaml.YAMLError as ye:
+            raise ConfigurationError('failed to parse config') from ye
+
+        for section in self._config:
+            # Subsequent definitions of the same service will overwrite the
+            # previous ones.
+            self._build_rules(section)
+
+    def _build_rules(self, service):
+        """Build the rules sets for classification, exclusion, enrichment and
+        routing for each service.
+
+        :param service: The service for which rule sets will be generated
+        :returns: None
+        """
+        if 'message' not in self._config[service]:
+            self._config[service]['message'] = {}
+
+        if 'title' not in self._config[service]:
+            self._config[service]['title'] = {}
+
+        self._classification_rules[service] = []
+        self._exclusion_rules[service] = []
+        self._enrichment_rules[service] = []
+        self._routing_rules[service] = []
+
+        self._build_classification_rules(service, 'message')
+        self._build_classification_rules(service, 'title')
+
+        self._build_exclusion_rules(service, 'message')
+        self._build_exclusion_rules(service, 'title')
+
+        self._build_enrichment_rules(service, 'message')
+        self._build_enrichment_rules(service, 'title')
+
+        self._build_routing_rules(service, 'message')
+        self._build_routing_rules(service, 'title')
+
+        # Ensure that required rule sets are defined
+        if not self._classification_rules:
+            raise ConfigurationError(f'classification rules not defined for {service}')
+
+        if not self._routing_rules:
+            raise ConfigurationError(f'routes not defined for {service}')
+
+    def _build_classification_rules(self, service, source):
+        """Build the classification rule set for a service
+
+        :param service: The service for which rule sets will be generated
+        :param source: The source field from the Alert object that will be used
+        :returns: None
+        """
+        service = service.lower()
+        cfg = self._config[service][source]
+
+        # Default to returning UNKNOWN severity
+        if 'classification' not in cfg:
+            self._classification_rules[service].append(lambda x: Severity.UNKNOWN)
+            return
+
+        self._classification_rules[service].append(
+            lambda x, cfg=cfg: Severity.CRITICAL if any(crit in getattr(x, source).lower() for crit in
```

Comment:
Agreed. I think as it grew I let my love for lambda go too far. I was drunk on ~~power~~ lambda.